agent/src/main/java/org/openremote/agent/protocol/tradfri/TradfriAsset.java

```diff
@@ -0,0 +1,62 @@
+package org.openremote.agent.protocol.tradfri;
+
+import org.openremote.agent.protocol.ProtocolAssetService;
+import org.openremote.agent.protocol.tradfri.device.Device;
+import org.openremote.container.util.UniqueIdentifierGenerator;
+import org.openremote.model.asset.Asset;
+import org.openremote.model.asset.AssetAttribute;
+import org.openremote.model.asset.AssetType;
+import org.openremote.model.attribute.MetaItem;
+
+import java.util.HashMap;
+
+public abstract class TradfriAsset {
+    private String name;
+    protected Asset asset;
+    protected AssetAttribute protocolConfiguration;
+    protected MetaItem agentLink;
+    protected ProtocolAssetService assetService;
+    protected Device device;
+
+    public TradfriAsset(AssetAttribute protocolConfiguration, MetaItem agentLink, ProtocolAssetService assetService, Device device, AssetType assetType, HashMap<String, Device> tradfriDevices) {
```

Comment:
Haven't dug into this but wondering why protocolConfiguration and assetService are being passed in here.
Why is this called an Asset....in OR we have an Asset class and this causes a bit of confusion.
What is the difference between TradfriAsset(s) and Tradfri device(s)

Reply:
I changed it so TradfriAsset now extends the Asset class of OpenRemote.
The protocol configuration is no longer passed, but the asset service is used by the asset to merge new values (when updates occur).